packages/turf-helpers/index.ts

```diff
@@ -207,30 +207,39 @@ export function feature<
  * var geometry = turf.geometry(type, coordinates);
  * // => geometry
  */
-export function geometry(
-  type:
+export function geometry<
+  T extends
     | "Point"
     | "LineString"
     | "Polygon"
     | "MultiPoint"
     | "MultiLineString"
     | "MultiPolygon",
+>(
+  type: T,
   coordinates: any[],
   _options: Record<string, never> = {}
-) {
+): {
+  Point: Point;
+  LineString: LineString;
+  Polygon: Polygon;
+  MultiPoint: MultiPoint;
+  MultiLineString: MultiLineString;
+  MultiPolygon: MultiPolygon;
+}[T] {
```

Comment:
I had considered making this a named interface and mapping against that instead (`GeometryTypeNameToType`), and referencing that instead of `as any` below, but I opted to not increase the public API for a mapping we only used in a single place.